From f859fed03d5d039430baea03a0ef35cfd6ac4abd Mon Sep 17 00:00:00 2001
From: Anssi Hannula <anssi.hannula@iki.fi>
Date: Thu, 6 Mar 2014 18:40:01 +0100
Subject: [PATCH 10/14] http: Always allow no-op seek

This also allows checking stream position as per ffurl_seek() doxy.

Signed-off-by: Luca Barbato <lu_zero@gentoo.org>
(cherry picked from commit ab76d9f628ad46e1d3bbf26c5bf1f87083f239ab)
---
 libavformat/http.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavformat/http.c b/libavformat/http.c
index 54910a9..2656595 100644
--- a/libavformat/http.c
+++ b/libavformat/http.c
@@ -901,6 +901,9 @@ static int64_t http_seek(URLContext *h, int64_t off, int whence)
 
     if (whence == AVSEEK_SIZE)
         return s->filesize;
+    else if ((whence == SEEK_CUR && off == 0) ||
+             (whence == SEEK_SET && off == s->off))
+        return s->off;
     else if ((s->filesize == -1 && whence == SEEK_END) || h->is_streamed)
         return AVERROR(ENOSYS);
 
-- 
1.7.10.4

